script.js

```diff
@@ -103,11 +107,13 @@ digitNextClear = false,
 evaluateNext = false,
 lastOperator, // For successive =
 lastNumber, // For successive =
+operatorSet = false,
 ans;
 
 const digits = document.querySelectorAll("button.digit"); // 0-9 and period
 digits.forEach(digit => {
     digit.addEventListener("click", () => {
+        operatorSet = false;
```

Comment:
Every time a digit is pressed, abort the -= possibility. e.g. - 2 = is not -= so abort, but -=, after the -, the operatorSet will become true, and this true controls the behavior of evaluateNow() triggered by = click. After that, the = will set the operatorSet to false again.